lib/admin.rb

```diff
@@ -0,0 +1,53 @@
+require 'pry'
+class Admin
+MAX_AVAILABILITY = 20
+  attr_reader :all_rooms, :all_reservations
+
+  def initialize
+    @all_rooms = (1..20).to_a
+    @all_reservations = []
+  end
+
+  def available_rooms(start_date,end_date)
+    all_available_rooms = all_rooms
```

Comment:
This makes `all_available_rooms` and `all_rooms` point to the same array in memory, thus deleting elements deletes to one delete from both.  Better to use `.dup` to duplicate the array.